system/lib/libc/musl/src/thread/__timedwait.c

```diff
@@ -41,7 +41,7 @@ int __timedwait_cp(volatile int *addr, int val,
 
 #ifdef __EMSCRIPTEN__
 	double msecsToSleep = top ? (top->tv_sec * 1000 + top->tv_nsec / 1000000.0) : INFINITY;
-	int is_main_thread = emscripten_is_main_browser_thread();
+	int is_main_thread = emscripten_is_main_runtime_thread();
```

Comment:
Is the behavior of this function actually supposed to be different on the non-browser thread, where we can't actually sleep? Or is that difference handled somewhere else?

Reply:
Oh, I think you're right, this change is incorrect. The logic here is actually checking whether we can do a synchronous wait or not, so it does care about the main browser thread.

The problem fixed in this PR is for proxying, not for sleeping. So this part is wrong and should not be needed. Verified the test passes without it locally.